addons/product/models/product_template.py

```diff
@@ -547,61 +547,21 @@ def _compute_display_name(self):
                 ))
 
     @api.model
-    def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
+    def _search_display_name(self, operator, value):
+        domain = super()._search_display_name(operator, value)
+        if self.env.context.get('search_product_product', bool(value)):
+            combine = expression.OR if operator not in expression.NEGATIVE_TERM_OPERATORS else expression.AND
+            domain = combine([domain, [('product_variant_ids', operator, value)]])
```

Comment:
Is this equivalent to the 50 lines it replaces ?
This looks like a simplification...